src/revlanguage/distributions/Dist_Inverse.h

```diff
@@ -0,0 +1,116 @@
+        Dist_Inverse* clone(void) const {
+            return new Dist_Inverse(*this);
+        }
+        
+        // Get Rev type
+        static const std::string& getClassType(void) {
+            static std::string rev_type = "Dist_Inverse";
+            return rev_type;
+        }
+        
+        // Get class type spec
+        static const TypeSpec& getClassTypeSpec(void) {
+            static TypeSpec rev_type_spec = TypeSpec( getClassType(), new TypeSpec( TypedDistribution< valType >::getClassTypeSpec() ) );
+            return rev_type_spec;
+        }
+   
+        // Get the alternative Rev names (aliases) for the constructor function
+        std::vector<std::string> getDistributionFunctionAliases( void ) const
+        {
+            // create alternative constructor function names variable that is the same for all instance of this class
+            std::vector<std::string> a_names;
+            a_names.push_back( "inv" );
+            
+            return a_names;
+        }
+
+        // Get the Rev-name for this distribution
+        std::string getDistributionFunctionName(void) const {
+            return "inverse";
+        }
+        
+        // Get the type spec of the instance
+        const TypeSpec& getTypeSpec(void) const {
+            static TypeSpec ts = getClassTypeSpec();
+            return ts;
+        }
+        
+        // Get member rules (const)
+        const MemberRules& getParameterRules(void) const {
+            static MemberRules dist_member_rules;
+            static bool rules_set = false;
+            
+            if ( !rules_set ) {
+                dist_member_rules.push_back( new ArgumentRule( "distribution", 
+                    TypedDistribution<valType>::getClassTypeSpec(), 
+                    "The distribution to invert.", 
+                    ArgumentRule::BY_CONSTANT_REFERENCE, 
+                    ArgumentRule::ANY ) );
+                rules_set = true;
+            }
+            
+            return dist_member_rules;
+        }
+        
+        // Distribution functions you have to override
+        RevBayesCore::InverseDistribution<typename valType::valueType>* createDistribution(void) const {
+            // get the parameters
+            const Distribution& orig_dist = static_cast<const Distribution &>( dist->getRevObject() );
+            
+            // Cast the single distribution to the specific type expected by InverseDistribution
+            RevBayesCore::TypedDistribution<typename valType::valueType>* typedDistPtr = 
+                static_cast<RevBayesCore::TypedDistribution<typename valType::valueType>* >( orig_dist.createDistribution() );
+
+            // Create an instance of InverseDistribution using the single distribution
+            RevBayesCore::InverseDistribution<typename valType::valueType>* d = 
+                new RevBayesCore::InverseDistribution<typename valType::valueType>(*typedDistPtr);
+
+            delete typedDistPtr;
```

Comment:
I don't understand why we do this - d still needs access to the base distribution, no ?

Reply:
My understanding is that the `InverseDistribution` constructor clones the distribution that it's passed; once it's been cloned, there's no further use for the intermediate `typedDistPtr`, which just served to cast the original distribution to the type expected by the constructor. `d` then works with a copy (clone) of `orig_dist`.

It's possible I'm misunderstanding something, though?

Reply:
ok that makes sense - I'm just a bit paranoid about `delete` because it's a good opportunity for segfaults